src/Bicep.Core.IntegrationTests/EvaluationTests.cs

```diff
@@ -258,5 +258,60 @@ public void Existing_resource_property_access_works()
                 evaluated.Should().HaveValueAtPath("$.outputs['abcVal'].value", "test!!!");
             }
         }
+
+        [TestMethod]
+        public void Items_function_evaluation_works()
+        {
+            var (template, _, _) = CompilationHelper.Compile(@"
+param inputObj object
+
+output inputObjKeys array = [for item in items(inputObj): item.key]
+output inputObjValues array = [for item in items(inputObj): item.value]
+");
+
+            using (new AssertionScope())
+            {
+                var evaluated = TemplateEvaluator.Evaluate(template, config => config with {
+                    Parameters = new()
+                    {
+                        ["inputObj"] = new JObject
+                        {
+                            ["ghiKey"] = "ghiValue",
+                            ["defKey"] = "defValue",
+                            ["abcKey"] = "abcValue",
+                            ["123Key"] = "123Value",
+                            ["GHIKey"] = "GHIValue",
+                            ["DEFKey"] = "DEFValue",
+                            ["ABCKey"] = "ABCValue",
+                            ["456Key"] = "456Value",
+                        }
+                    }
+                });
+
+                evaluated.Should().HaveValueAtPath("$.outputs['inputObjKeys'].value", new JArray
+                {
```

Comment:
Why the output is sorted? I’d expect it’ll be in order that it appears in the iterated object.

Reply:
See https://github.com/Azure/bicep/issues/1427#issuecomment-920049749

Reply:
gotcha!